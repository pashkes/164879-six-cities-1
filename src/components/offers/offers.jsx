import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import CardPlace from '../card-place/card-place.jsx';
import {connect} from "react-redux";

class Offers extends PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      selected: null
    };
    this.onSelected = this.onSelected.bind(this);
  }

  onSelected(id) {
    this.setState({current: id});
  }

  render() {
    const offers = this.props.offers;
    return <div className="cities__places-list places__list tabs__content">
      {offers.map((offer) => {
        return <CardPlace
          key={offer.id}
          {...offer}
          onSelected={this.onSelected}
          selected={this.state.currentCity === offer.id}
        />;
      })}
    </div>;
  }
}

Offers.propTypes = {
  offers: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    photoURL: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
    isPremium: PropTypes.bool.isRequired,
    isFavorite: PropTypes.bool.isRequired,
    price: PropTypes.number.isRequired,
    rating: PropTypes.number.isRequired,
  })).isRequired,
};

const mapStateToProps = (state) => {
  return {
    offers: state.offers.filter((offer) => {
      return offer.city === state.city;
    })
  };
};

export default connect(mapStateToProps)(Offers);
