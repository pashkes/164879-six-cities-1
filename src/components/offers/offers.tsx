import * as React from "react";
import {connect} from "react-redux";

import {ActionCreators} from "./../../reducer/data/data";
import CardPlace from "../card-place/card-place";

import {Offer} from "../../types";

interface Props {
  offers: Offer[],
  classModOffers: string,
  classModCard: string,
  onSetActiveOffer: (id: number) => void,
}

export class Offers extends React.PureComponent<Props> {
  onSelectOffer = (id) => {
    this.props.onSetActiveOffer(id);
  };

  render() {
    const {offers, classModOffers = []} = this.props;

    return <div className={`${classModOffers} places__list`}>
      {offers.map((offer) => {
        return <CardPlace
          key={offer.id}
          {...offer}
          onSelected={this.onSelectOffer}
          classModPrefix={`cities`}
          mainClassMod={`cities__place-card`}
        />;
      })}
    </div>;
  }
}

const mapDispatchToProps = (dispatch) => ({
  onSetActiveOffer: (id) => dispatch(ActionCreators.setActiveIdOffer(id)),
});

export default connect(null, mapDispatchToProps)(Offers);
