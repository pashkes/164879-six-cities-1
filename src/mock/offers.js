const offers = [
  {
    id: `5454`,
    city: `Amsterdam`,
    title: `Beautiful & luxurious apartment at great location`,
    photoURL: `img/apartment-01.jpg`,
    isPremium: true,
    isFavorite: false,
    price: 120,
    type: `Apartment`,
    rating: 93,
    coordinates: [52.3909553943508, 4.85309666406198]
  },
  {
    id: `5675675`,
    city: `Amsterdam`,
    title: `Wood and stone place`,
    photoURL: `img/room.jpg`,
    isPremium: false,
    isFavorite: true,
    price: 80,
    type: `Private room`,
    rating: 80,
    coordinates: [52.369553943508, 4.85309666406198]
  },
  {
    id: `545345`,
    city: `Amsterdam`,
    title: `Canal View Prinsengracht`,
    photoURL: `img/apartment-02.jpg`,
    isPremium: false,
    isFavorite: false,
    price: 132,
    type: `Apartment`,
    rating: 80,
    coordinates: [52.3909553943508, 4.929309666406198]
  },
  {
    id: `878974`,
    city: `Amsterdam`,
    title: `Nice, cozy, warm big bed apartment`,
    photoURL: `img/apartment-03.jpg`,
    isPremium: false,
    isFavorite: false,
    price: 180,
    type: `Apartment`,
    rating: 80,
    coordinates: [52.3809553943508, 4.939309666406198]
  },
  {
    id: `fdfdg`,
    city: `Paris`,
    title: `Beautiful & luxurious apartment at great location`,
    photoURL: `img/apartment-01.jpg`,
    isPremium: true,
    isFavorite: false,
    price: 120,
    type: `Apartment`,
    rating: 93,
    coordinates: [52.3909553943508, 4.85309666406198]
  },
  {
    id: `jfghjf`,
    city: `Cologne`,
    title: `Wood and stone place`,
    photoURL: `img/room.jpg`,
    isPremium: false,
    isFavorite: true,
    price: 80,
    type: `Private room`,
    rating: 80,
    coordinates: [52.369553943508, 4.85309666406198]
  },
  {
    id: `jjfghjgf56`,
    city: `Hamburg`,
    title: `Canal View Prinsengracht`,
    photoURL: `img/apartment-02.jpg`,
    isPremium: false,
    isFavorite: false,
    price: 132,
    type: `Apartment`,
    rating: 80,
    coordinates: [52.3909553943508, 4.929309666406198]
  },
  {
    id: `bncvb`,
    city: `Dusseldorf`,
    title: `Nice, cozy, warm big bed apartment`,
    photoURL: `img/apartment-03.jpg`,
    isPremium: false,
    isFavorite: false,
    price: 180,
    type: `Apartment`,
    rating: 80,
    coordinates: [52.3809553943508, 4.939309666406198]
  }
];
export default offers;
