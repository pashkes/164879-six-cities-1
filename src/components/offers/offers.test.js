import React from 'react';
import {Offers} from "./../offers/offers";
import offers from "../../mock/offers";
import {shallow, configure} from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({adapter: new Adapter()});

it(`list of offers renders correctly`, () => {
  const cards = shallow(<Offers offers={offers} />);
  expect(cards).toMatchSnapshot();
});
