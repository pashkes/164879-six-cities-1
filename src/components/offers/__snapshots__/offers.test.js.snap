// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list of offers renders correctly 1`] = `
<div
  className=" places__list"
>
  <CardPlace
    city="Amsterdam"
    classModPrefix="cities"
    coordinates={
      Array [
        52.3809553943508,
        4.939309666406198,
      ]
    }
    id={334454}
    isFavorite={false}
    isPremium={false}
    key="334454"
    mainClassMod="cities__place-card"
    onSelected={[Function]}
    previewPhoto="img/apartment-03.jpg"
    price={180}
    rating={80}
    title="Nice, cozy, warm big bed apartment"
    type="Apartment"
  />
  <CardPlace
    city="Amsterdam"
    classModPrefix="cities"
    coordinates={
      Array [
        52.3909553943508,
        4.85309666406198,
      ]
    }
    id={5435}
    isFavorite={false}
    isPremium={true}
    key="5435"
    mainClassMod="cities__place-card"
    onSelected={[Function]}
    previewPhoto="img/apartment-01.jpg"
    price={120}
    rating={93}
    title="Beautiful & luxurious apartment at great location"
    type="Apartment"
  />
  <CardPlace
    city="Amsterdam"
    classModPrefix="cities"
    coordinates={
      Array [
        52.369553943508,
        4.85309666406198,
      ]
    }
    id={54353}
    isFavorite={true}
    isPremium={false}
    key="54353"
    mainClassMod="cities__place-card"
    onSelected={[Function]}
    previewPhoto="img/room.jpg"
    price={80}
    rating={80}
    title="Wood and stone place"
    type="Private room"
  />
  <CardPlace
    city="Amsterdam"
    classModPrefix="cities"
    coordinates={
      Array [
        52.3909553943508,
        4.929309666406198,
      ]
    }
    id={53453}
    isFavorite={false}
    isPremium={false}
    key="53453"
    mainClassMod="cities__place-card"
    onSelected={[Function]}
    previewPhoto="img/apartment-02.jpg"
    price={132}
    rating={80}
    title="Canal View Prinsengracht"
    type="Apartment"
  />
  <CardPlace
    city="Paris"
    classModPrefix="cities"
    coordinates={
      Array [
        48.86679,
        2.37852,
      ]
    }
    id={534534}
    isFavorite={false}
    isPremium={true}
    key="534534"
    mainClassMod="cities__place-card"
    onSelected={[Function]}
    previewPhoto="img/apartment-01.jpg"
    price={120}
    rating={93}
    title="Beautiful & luxurious apartment at great location"
    type="Apartment"
  />
  <CardPlace
    city="Cologne"
    classModPrefix="cities"
    coordinates={
      Array [
        50.958694,
        6.961092,
      ]
    }
    id={3434}
    isFavorite={true}
    isPremium={false}
    key="3434"
    mainClassMod="cities__place-card"
    onSelected={[Function]}
    previewPhoto="img/room.jpg"
    price={80}
    rating={80}
    title="Wood and stone place"
    type="Private room"
  />
  <CardPlace
    city="Hamburg"
    classModPrefix="cities"
    coordinates={
      Array [
        53.598154,
        10.062009,
      ]
    }
    id={65656}
    isFavorite={false}
    isPremium={false}
    key="65656"
    mainClassMod="cities__place-card"
    onSelected={[Function]}
    previewPhoto="img/apartment-02.jpg"
    price={132}
    rating={80}
    title="Canal View Prinsengracht"
    type="Apartment"
  />
  <CardPlace
    city="Dusseldorf"
    classModPrefix="cities"
    coordinates={
      Array [
        51.261306,
        6.804574,
      ]
    }
    id={23232}
    isFavorite={false}
    isPremium={false}
    key="23232"
    mainClassMod="cities__place-card"
    onSelected={[Function]}
    previewPhoto="img/apartment-03.jpg"
    price={180}
    rating={80}
    title="Nice, cozy, warm big bed apartment"
    type="Apartment"
  />
</div>
`;
